ts/src/main/scala/ch/epfl/ts/traders/Trader.scala

```diff
@@ -1,14 +1,32 @@
 package ch.epfl.ts.traders
 
+import scala.reflect.ClassTag
+
+import akka.actor.Props
 import ch.epfl.ts.component.Component
+import ch.epfl.ts.component.ComponentBuilder
+import ch.epfl.ts.component.ComponentRef
 import ch.epfl.ts.data.ParameterTrait
+import ch.epfl.ts.data.StrategyParameters
+import scala.reflect
+
+case class RequiredParameterMissingException(message: String) extends RuntimeException(message)
 
 /**
  * Abstract class to be extended by all concrete Trader implementations.
  * A Trader represents the implementation of a trading strategy.
+ * 
+ * It factorizes parameter handling for concrete trading strategies.
+ * 
+ * It will throw a `RequiredParmaeterMissingException` on instantiation if any of the
+ * required parameters have not been provided (or have the wrong type).
  */
-abstract class Trader extends Component {
+abstract class Trader(parameters: StrategyParameters) extends Component {
+  /** Gives a handle to the companion object */
+  def companion: TraderCompanion
   
+  // On instantiation, check that all mandatory parameters have been provided
```

Comment:
What does it do? What is the difference between this and `getInstance`? As far as I understood you wanted to throw the exception when some params are not provided/wrong, so why do you need getInstance then?